jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -751,6 +751,75 @@ ecma_get_named_data_property (ecma_object_t *obj_p, /**< object to find property
   return ECMA_PROPERTY_VALUE_PTR (property_p);
 } /* ecma_get_named_data_property */
 
+/**
+ * Release the namedata property value and invalidate it from Lcache
+ */
+inline void JERRY_ATTR_ALWAYS_INLINE
+ecma_release_nameddata_property_value (ecma_object_t *object_p, /**< object the property belongs to */
+                                       ecma_string_t *name_p, /**< name of the property */
+                                       ecma_property_t *property_p) /**< property */
+{
+  JERRY_ASSERT (object_p != NULL && property_p != NULL);
+  JERRY_ASSERT (ECMA_PROPERTY_GET_TYPE (*property_p) == ECMA_PROPERTY_TYPE_NAMEDDATA);
+  JERRY_UNUSED_2 (object_p, name_p);
+
+  ecma_free_value_if_not_object (ECMA_PROPERTY_VALUE_PTR (property_p)->value);
+
+#if ENABLED (JERRY_LCACHE)
+  jmem_cpointer_t name_cp;
+  if (ECMA_IS_DIRECT_STRING (name_p))
+  {
+    name_cp = (jmem_cpointer_t) ECMA_GET_DIRECT_STRING_VALUE (name_p);
+  }
+  else
+  {
+    ECMA_SET_NON_NULL_POINTER (name_cp, name_p);
+  }
+
+  if (ecma_is_property_lcached (property_p))
+  {
+    ecma_lcache_invalidate (object_p, name_cp, property_p);
+  }
+#endif /* ENABLED (JERRY_LCACHE) */
+} /* ecma_release_nameddata_property_value */
+
+/**
+ * Release the accessor property value and invalidate it from Lcache
+ */
+inline void JERRY_ATTR_ALWAYS_INLINE
+ecma_release_accessor_property_value (ecma_object_t *object_p, /**< object the property belongs to */
+                                      ecma_string_t *name_p, /**< name of the property */
+                                      ecma_property_t *property_p) /**< property */
+{
+  JERRY_ASSERT (object_p != NULL && property_p != NULL);
+  JERRY_ASSERT (ECMA_PROPERTY_GET_TYPE (*property_p) == ECMA_PROPERTY_TYPE_NAMEDACCESSOR);
+  JERRY_UNUSED_3 (object_p, name_p, property_p);
+
+#if ENABLED (JERRY_CPOINTER_32_BIT)
+  ecma_getter_setter_pointers_t *getter_setter_pair_p;
+  getter_setter_pair_p = ECMA_GET_NON_NULL_POINTER (ecma_getter_setter_pointers_t,
+                                                    ECMA_PROPERTY_VALUE_PTR (property_p)->getter_setter_pair_cp);
+  jmem_pools_free (getter_setter_pair_p, sizeof (ecma_getter_setter_pointers_t));
+#endif /* ENABLED (JERRY_CPOINTER_32_BIT) */
+
+#if ENABLED (JERRY_LCACHE)
+  jmem_cpointer_t name_cp;
+  if (ECMA_IS_DIRECT_STRING (name_p))
+  {
+    name_cp = (jmem_cpointer_t) ECMA_GET_DIRECT_STRING_VALUE (name_p);
+  }
+  else
+  {
+    ECMA_SET_NON_NULL_POINTER (name_cp, name_p);
+  }
+
+  if (ecma_is_property_lcached (property_p))
+  {
+    ecma_lcache_invalidate (object_p, name_cp, property_p);
```

Comment:
Why do we need to invalidate lcache? Isn't hashing only use object_p and name_p? Those will not change.

Reply:
Indeed!